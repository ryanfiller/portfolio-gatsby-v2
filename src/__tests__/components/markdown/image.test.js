import React from 'react'
import Image from '../../../components/markdown/image'

describe('<Image />', () => {
  const props = () => ({
    src: 'string.jpg',
    alt: 'alt',
    title: null,
    'data-align': null,
    'data-small': null,
    'data-caption': null
  })

  it('should render correctly', async () => {
    const component = mount(<Image src='test.jpg' alt='alt' title='title' />)      
    expect(component.html()).toMatchSnapshot()
    expect(await axe(component.html())).toHaveNoViolations()
  })

  describe('which tag to render', () => {
    it('should render an <img> by default', async () => {
      const component = mount(<Image {...props()} />)
      expect(component.find('img').length).toEqual(1)
      expect(component.find('figure').length).toEqual(0)
      expect(await axe(component.html())).toHaveNoViolations()
    })

    it('should render an <figure> and <figcaption> if there is a caption', async () => {
      const component = mount(<Image {...props()} data-caption='caption' />)
      expect(component.find('figure').length).toEqual(1)
      expect(component.find('figcaption').length).toEqual(1)
      expect(component.find('figcaption').text()).toEqual('caption')
      expect(await axe(component.html())).toHaveNoViolations()
    })
  })

  describe('transformation params', () => {
    it('should not transform a .gif', () => {
      const component = mount(<Image {...props()} src='text.gif' />)
      expect(component.find('img').props().src).toEqual('text.gif')
    })

    it('should not transform a .svg', () => {
      const component = mount(<Image {...props()} src='text.svg' />)
      expect(component.find('img').props().src).toEqual('text.svg')
    })

    it('should transform images and add a srcSet', async () => {
      const component = mount(<Image {...props()} />)
      expect(component.find('img').props().src.includes('?nf_resize=fit&w=')).toEqual(true)
      expect(component.find('img').props().srcSet.includes('?nf_resize=fit&w=')).toEqual(true)
      expect(await axe(component.html())).toHaveNoViolations()
    })
  })

  describe('alignment and sizes', () => {
    it('render a small image', () => {
      const imgComponent = mount(<Image {...props()} data-small='true' />)
      expect(imgComponent.find('img.image--small').length).toEqual(1)
      const figComponent = mount(<Image {...props()} data-small='true' data-caption='caption' />)
      expect(figComponent.find('img.image--small').length).toEqual(0)
      expect(figComponent.find('figure.image--small').length).toEqual(1)
    })
  })
})