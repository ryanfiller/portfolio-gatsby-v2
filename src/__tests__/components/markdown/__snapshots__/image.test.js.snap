// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render correctly 1`] = `
"<img class=\\"image image--undefined\\" alt=\\"alt\\" title=\\"title\\" srcset=\\"
            test.jpg?nf_resize=fit&amp;w=500 500w, 
            test.jpg?nf_resize=fit&amp;w=800 8000w
          \\" sizes=\\"100vw\\" src=\\"test.jpg?nf_resize=fit&amp;w=1000\\">"
`;
